### PR TITLE
[BEAM-1062] Shade the SDK more fully; everything but a whitelist

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

Rebased on tons of changes, including a possible fix to the xerial issue.

### CODE REVIEW

**Comment by kennknowles** [comment]
Hmm, rebase fail. I'll see if our new pom configuration is compatible with going all the way on this.

File: sdks/java/core/pom.xml [hunk 1]
```diff
@@ -61,7 +61,31 @@
...
+           - If a dependency is merely used by the SDK as a private implementation
+             detail, the class is included in a relocated form so it will
+             never cause a conflict with any other user or library's
+             dependencies.
+
+          In the first execution, all of the relocated dependencies are
+          bundled in a minimized jar. Because they are an implementation
+          detail, this should not be observable.
+
+          In the second execution, all of the dependencies that are part of
```
Comment (kennknowles) [inline sdks/java/core/pom.xml]: I believe that when this effort began, we (rather impolitely) bundled all of our non-relocated dependencies so our jar had a bunch of stuff outside the namespaces that Beam owns. Now I think we are doing things in a more standard way so the second execution can be wiped out.

**Comment by davorbonaci** [comment]
Obsolete? Close?

**Comment by kennknowles** [comment]
Not obsolete at all. This is important work that just didn't quite make it to top priority.

**Comment by kennknowles** [comment]
It appears the root configuration corrupts child configurations if they switch from blacklist to whitelist. There have been some fixes to our snappy dependency that should make this go right through once I fix that.

**Comment by reuvenlax** [comment]
@kennknowles what is the status of this PR?

**Comment by kennknowles** [comment]
There are both issues where our use of inheritance is broken and also a couple outstanding dependencies that are trouble. Beyond that, there's the bloat problem which is looking like it might be a real problem nowadays, whereas before it was premature optimization to worry about.

**Pull request closed without merging**